### ISSUE TITLE
Questions about the training and evaluation procedure.

### ISSUE THREAD

**Issue by Hsu9612** [chensong1995/HybridPose#7]
Hi Chen, 
thanks for sharing the work. I have some questions while reading your code:

1. I download the augmented labels and see the valid poses you provided. The number of the valid poses is a lot less than the number of gt poses (1214) provided by the original Occlusion LineMOD. I wonder how you got these valid poses and why you use these valid poses for evaluation instead of the gt poses provided by the original Occlusion LineMOD?

2. In [setup_loaders()](https://github.com/chensong1995/HybridPose/blob/83513cef0d977bf1ad90f17ef8a0b16039164b9f/src/train_core.py#L48), I see you randomly select 80% of data in Occlusion LineMOD as the training set and the other 20% as the test set. In [generate_data()](https://github.com/chensong1995/HybridPose/blob/83513cef0d977bf1ad90f17ef8a0b16039164b9f/trainers/coretrainer.py#L530) you search for parameters with the first 20 samples of the test set and evaluate with the remaining samples in the test set. Am I right? Please correct me if my understanding is wrong.

I will be looking forward to your reply.

**Comment by chensong1995** [comment]
Hello Hsu9612, 

Thank you for your questions!

1. An example in the original Occlussion Linemod dataset does not necessarily include all objects. For each object, the number of valid poses should match the number of Occlussion Linemod examples that contain this object. 

2. Your are right.

**Comment by Hsu9612** [comment]
Thanks for your prompt reply!

Could you tell me how to define a valid pose. Because I can see ape in 00023.png but it was removed in your valid pose (although the number of visible pixels is very small).

Models that run on OccludedLineMOD are usually trained with synthetic data or real data from Linemod. Whether the results obtained by training with OcclusionLineMOD data are biased

**Comment by chensong1995** [comment]
Hello Hsu9612,

I saw your comment yesterday. When I was typing my response, your comment disappeared. Did I accidentally delete your comment? Or do you need some more time editing?

**Comment by Hsu9612** [comment]
Actually I deleted it myself.
I still want to know how your valid pose is defined cause I can see ape in 00023.png but it was removed in your valid pose (although the number of visible pixels is very small). In addition, I learned in [Krull's work](https://www.cv-foundation.org/openaccess/content_iccv_2015/papers/Krull_Learning_Analysis-by-Synthesis_for_ICCV_2015_paper.pdf) that there were 8992 test cases in OcclusionLineMOD. I wonder if there is a unified evaluation protocol on it.

**Comment by chensong1995** [comment]
> Actually I deleted it myself.
> I still want to know how your valid pose is defined cause I can see ape in 00023.png but it was removed in your valid pose (although the number of visible pixels is very small). In addition, I learned in [Krull's work](https://www.cv-foundation.org/openaccess/content_iccv_2015/papers/Krull_Learning_Analysis-by-Synthesis_for_ICCV_2015_paper.pdf) that there were 8992 test cases in OcclusionLineMOD. I wonder if there is a unified evaluation protocol on it.

- We use the segmentation mask computed by the authors of [PVNet](https://github.com/zju3dv/pvnet). The mask of example 00023 does not contain any pixel that is part of the ape. Therefore, this example is removed. The shell script I used to create valid poses is:

```shell
names=('ape' 'can' 'cat' 'driller' 'duck' 'eggbox' 'glue' 'holepuncher')
mkdir -p valid_poses
for name in "${names[@]}"; do
    mkdir -p valid_poses/$name
    new_idx=0
    for filename in poses/$name/*.txt; do
        line_count="$(cat $filename | wc -l)"
        if [ $line_count -gt 2 ]
        then
            old_idx="$(basename $filename .txt | cut -c 6-)"
            old_idx="$(echo $old_idx | sed 's/^0*//')"
            old_idx_len="$(echo $old_idx | wc -c)"
            if [ $old_idx_len -eq 1 ]
            then
                old_idx=0
            fi
            mask_count="$(python mask_count.py masks/$name/$old_idx.png)"
            if [ $mask_count -gt 0 ]
            then
                cp $filename valid_poses/$name/$new_idx.txt
                echo -e "\n" >> valid_poses/$name/$new_idx.txt
                basename $filename .txt | cut -c 6- >> valid_poses/$name/$new_idx.txt
                new_idx=$((new_idx + 1))
            else
                echo $filename' has empty mask, filtered'
            fi
        else
            echo $filename' contains '$line_count' lines, filtered'
        fi
    done
done
```

- Both PVNet and HybridPose download the Occlusion Linemod dataset from [here](https://hci.iwr.uni-heidelberg.de/vislearn/iccv2015-occlusion-challenge/#Dataset). I am aware that there seems to be an (unreleased) test set on Occlusion Linemod in the original ICCV 2015 challenge. However, I do not have access to that private test set, and could only evaluate HybridPose on a random split of publicly available examples.

I hope this helps.

**Comment by Hsu9612** [comment]
Thanks for your prompt reply! This helps me a lot.
